### ISSUE TITLE
Since 10.0.1, saving an image as a JPEG does not always result in files of the same size

### ISSUE THREAD

**Issue by johnnyleitrim** [python-pillow/Pillow#7477]
### What did you do?
Using Pillow 10.0.1 and above, saving an image as a JPEG does not always result in the same number of bytes being written.

### What did you expect to happen?
Saving the same image multiple times as a JPEG would result in files of the same size.

### What actually happened?
The file sizes differ

### What are your OS, Python and Pillow versions?

* OS: macOS Ventura 13.6
* Python: 3.9.6
* Pillow: 10.0.1

```python
from io import BytesIO

from PIL import Image

image = Image.new(mode="RGB", size=(1024, 1024), color="red")

for _ in range(10):
    output = BytesIO()
    image.save(output, format="jpeg")
    output.seek(0)
    print(len(output.read()))
```

#### Actual Output:
```
17013
17014
17014
17014
17014
17013
17013
17014
17014
17014
```

### Expected output:
The number of bytes should be the same for each iteration.

**Comment by radarhere** [comment]
How did you install Pillow?

Out of curiosity - how did you find this problem?

**Comment by johnnyleitrim** [comment]
@radarhere I installed it with `pip install pillow==10.0.1`

We found it because we have some integration tests in our product that create JPEGs, and we check the file size of the created JPEG. After upgrading, we saw that we were getting different file sizes for different executions.

**Comment by radarhere** [comment]
In Pillow 10.0.1, we [upgraded from libjpeg-turbo 2.1.5.1 to 3.0.0](https://github.com/python-pillow/pillow-wheels/pull/393). Testing, I find that upgrading libjpeg-turbo 3.0.1 fixes this.

So this is resolved by #7472

If you would like to test this, or have an immediate fix, you can install 
[Pillow-10.2.0.dev0-cp39-cp39-macosx_11_0_arm64.whl.zip](https://github.com/python-pillow/Pillow/files/13034204/Pillow-10.2.0.dev0-cp39-cp39-macosx_11_0_arm64.whl.zip). Just unzip it, uninstalling your current Pillow and run `python3 -m pip install Pillow-10.2.0.dev0-cp39-cp39-macosx_11_0_arm64.whl`

**Comment by johnnyleitrim** [comment]
@radarhere Do you know when 10.2.0 will be officially released?

**Comment by aclark4life** [comment]
Jan 1?

**Comment by hugovk** [comment]
Close, Jan 2 :)

https://github.com/python-pillow/Pillow/milestone/57

> Pillow is released quarterly on January 2nd, April 1st, July 1st and October 15th.

https://pillow.readthedocs.io/en/stable/releasenotes/

**Comment by johnnyleitrim** [comment]
From our testing, we can see that Pillow corrupts the last few pixels in the JPEGs, and that's why the file sizes are different. Is there any possibility of an earlier release?

**Comment by devbanu** [comment]
I encountered the same problem in a different way: on macOS the hardware JPEG decoder started complaining when decoding JPEG files created with Pillow 10.1.0.

The error loos like this:
`JPEGDecompressSurface : Picture decode failed: e00002e9`

**Comment by radarhere** [comment]
Pillow 10.2.0 has now been published, with libjpeg-turbo 3.0.1 in the wheels.

**Comment by aclark4life** [comment]
Also woo hoo!
